tools/pr-notify/github.py

```
@@ -0,0 +1,176 @@
+    except subprocess.TimeoutExpired:
+        raise SystemExit("GitHub GraphQL query timed out after 60 seconds")
+
+    # gh CLI returns non-zero on GraphQL errors but may still include
+    # partial data in stdout (e.g., one repo not found but others succeed).
+    # Try to parse JSON first before treating as a hard failure.
+    try:
+        response = json.loads(result.stdout)
+    except (json.JSONDecodeError, ValueError):
+        if result.returncode != 0:
+            error_msg = result.stderr.strip() or result.stdout.strip()
+            raise SystemExit(f"GitHub GraphQL query failed: {error_msg}")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)